src/nv_ingest/modules/sinks/vdb_task_sink.py

```diff
@@ -34,6 +39,61 @@
 
 VDBTaskSinkLoaderFactory = ModuleLoaderFactory(MODULE_NAME, MODULE_NAMESPACE, VdbTaskSinkSchema)
 
+_DEFAULT_ENDPOINT = os.environ.get("MINIO_INTERNAL_ADDRESS", "minio:9000")
+_DEFAULT_BUCKET_NAME = os.environ.get("MINIO_BUCKET", "nv-ingest")
+
+
+def _bulk_ingest(milvus_uri: str = None,
+                collection_name: str = None, 
+                bucket_name: str = None, 
+                bulk_ingest_path: str = None, 
+                extra_params: dict = None):
+
+    endpoint = extra_params.get("endpoint", _DEFAULT_ENDPOINT)
+    access_key = extra_params.get("access_key", None)
+    secret_key = extra_params.get("secret_key", None)
+
+    client = Minio(
+        endpoint,
+        access_key=access_key,
+        secret_key=secret_key,
+        session_token=extra_params.get("session_token", None),
+        secure=extra_params.get("secure", False),
+        region=extra_params.get("region", None),
+    )
+    bucket_found = client.bucket_exists(bucket_name)
+    if not bucket_found:
+        raise ValueError(f"Could not find bucket {bucket_name}")
+    batch_files = [[f"{file.object_name}"] for file in client.list_objects(bucket_name, prefix=bulk_ingest_path)]
+
+    uri_parsed = urlparse(milvus_uri)
+    conn = connections.connect(host=uri_parsed.hostname, port=uri_parsed.port)
+    collection = Collection(name=collection_name)
+
+    task_ids = []
+    for file in batch_files:
+        task_id = utility.do_bulk_insert(collection_name=collection_name, files=file)
+        task_ids.append(task_id)
+
+    while len(task_ids) > 0:
+        logger.info("Wait 1 second to check bulkinsert tasks state...")
+        time.sleep(1)
+        for id in task_ids:
+            state = utility.get_bulk_insert_state(task_id=id)
+            if state.state == BulkInsertState.ImportFailed or state.state == BulkInsertState.ImportFailedAndCleaned:
+                logger.error(f"The task {state.task_id} failed, reason: {state.failed_reason}")
+                task_ids.remove(id)
+            elif state.state == BulkInsertState.ImportCompleted:
+                logger.info(f"The task {state.task_id} completed")
+                task_ids.remove(id)
+
+    while True:
```

Comment:
That is going to be difficult because the amount of time an index can take to complete creation is entirely dependent on the number of entries ingested. I can add a count for the timeframe between updates but other than that we wont know how much time this will actually take. It will be very different for 1k records and 1 million records or 1 billion records.  Depending on the number of records this can take hours, and it might seem long but still orders of magnitude faster than CPU based upload.